## Homework 4

### Dua Batool
### db07098@st.habib.edu.pk

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats

In [2]:
df = pd.read_csv('EmployeeAttrition.csv')

In [3]:
df.head()

Age     BusinessTravel  DailyRate              Department  \
0   41      Travel_Rarely       1102                   Sales   
1   49  Travel_Frequently        279  Research & Development   
2   37      Travel_Rarely       1373  Research & Development   
3   33  Travel_Frequently       1392  Research & Development   
4   27      Travel_Rarely        591  Research & Development   

   DistanceFromHome  Education EducationField  EnvironmentSatisfaction  \
0                 1          2  Life Sciences                        2   
1                 8          1  Life Sciences                        3   
2                 2          2          Other                        4   
3                 3          4  Life Sciences                        4   
4                 2          1        Medical                        1   

   Gender  HourlyRate  ...  StandardHours  StockOptionLevel TotalWorkingYears  \
0  Female          94  ...             80                 0                 8   
1    Male          61  ...             80                 1                10   
2    Male          92  ...             80                 0                 7   
3  Female          56  ...             80                 0                 8   
4    Male          40  ...             80                 1                 6   

   TrainingTimesLastYear WorkLifeBalance  YearsAtCompany  YearsInCurrentRole  \
0                      0               1               6                   4   
1                      3               3              10                   7   
2                      3               3               0                   0   
3                      3               3               8                   7   
4                      3               3               2                   2   

   YearsSinceLastPromotion YearsWithCurrManager  Attrition  
0                        0                    5        Yes  
1                        1                    7         No  
2                        0                    0        Yes  
3                        3                    0         No  
4                        2                    2         No  

[5 rows x 32 columns]

### Question 1:

H0 (Null Hypothesis) = The monthly income of males is less than or equal to females.

H1 (Alternative Hypothesis) = The monthly income of males is greater than females.

In [4]:
male = df[(df['Gender']=='Male')]['MonthlyIncome']
female = df[(df['Gender']=='Female')]['MonthlyIncome']

In [5]:
stats.ttest_ind(male, female, alternative='greater')

Ttest_indResult(statistic=-1.2212617308870655, pvalue=0.8889084827245606)

The pvalue is greater than 0.05 so we would NOT reject the Null Hypothesis. 

The monthly income of males is less than or equal to females. We can justify this by saying that there is no gender discrimination in the workplace because of which the males are not getting paid more than the females. 

### Question 2

H0 (Null Hypothesis) = The WorkLifeBalance of males is greater than or equal to females.

H1 (Alternative Hypothesis) = The WorkLifeBalance of males is less than females.

In [6]:
male = df[(df['Gender']=='Male')]['WorkLifeBalance']
female = df[(df['Gender']=='Female')]['WorkLifeBalance']

In [7]:
stats.ttest_ind(male, female, alternative='less')

Ttest_indResult(statistic=-0.10546787956064164, pvalue=0.45800948246831286)

The pvalue is greater than 0.05 so we would NOT reject the Null Hypothesis.

The WorkLifeBalance of males is greater than or equal to females. We can justify this by saying that males are better at managing both work and life than females as generally the males do not have other responsibilities (exculding work) like managing the household which the females have to do.

### Question 3

H0 (Null Hypothesis) = The YearsAtCompany of Single is greater than or equal to married.

H1 (Alternative Hypothesis) = The YearsAtCompany of Single is less than married.

In [8]:
single = df[(df['MaritalStatus']=='Single')]['YearsAtCompany']
married = df[(df['MaritalStatus']=='Married')]['YearsAtCompany']

In [9]:
stats.ttest_ind(single, married, alternative='less')

Ttest_indResult(statistic=-2.5589985737464476, pvalue=0.0053127663549315815)

The pvalue is less than 0.05 so we would reject the Null Hypothesis.

The YearsAtCompany of Single is less than married. We can justify this by saying that the married prefer to remain at the same company while the singles prefer to shift to other companies or jobs.

### Question 4

H0 (Null Hypothesis) = The EnvironmentalSatisfaction of Attrition = Yes is greater than or equal to Attrition = No

H1 (Alternate Hypothesis) = The EnvironmentalSatisfaction of Attrition = Yes is less than Attrition = No

In [10]:
yes = df[(df['Attrition']=='Yes')]['EnvironmentSatisfaction']
no = df[(df['Attrition']=='No')]['EnvironmentSatisfaction']

In [11]:
stats.ttest_ind(yes, no, alternative='less')

Ttest_indResult(statistic=-3.9818600026818496, pvalue=3.58616927468443e-05)

The pvalue is less than 0.05 so we would reject the Null Hypothesis.

The EnvironmentalSatisfaction of Attrition = Yes is less than Attrition = No. We can justify this by saying that the employees who left the company are not as satisfied with environment as the employees who didn't.

### Question 5

H0 (Null Hypothesis) = The MonthlyIncome of Manager is less than or equal to Laboratory Technician

H1 (Alternate Hypothesis) = The MonthlyIncome of Manager is greater than Laboratory Technician

In [12]:
manager = df[(df['JobRole']=='Manager')]['MonthlyIncome']
tech = df[(df['JobRole']=='Laboratory Technician')]['MonthlyIncome']

In [13]:
stats.ttest_ind(manager, tech, alternative='greater')

Ttest_indResult(statistic=76.03317414634023, pvalue=9.971168334368433e-224)

The pvalue is less than 0.05 so we would reject the Null Hypothesis.

The MonthlyIncome of Manager is greater than Laboratory Technician. We can justify this by saying that the company values the managers more than the laboratory technicians as the managers are being paid more.

### Question 6

H0 (Null Hypothesis) = The Department and Gender columns are not associated with each other (independent)

H1 (Alternate Hypothesis) = The Department and Gender columns are associated with each other (not independent)

In [16]:
ct = pd.crosstab(df['Department'],df['Gender'], margins=True)   # creating contingency table

In [17]:
ct

Gender                  Female  Male   All
Department                                
Human Resources             20    43    63
Research & Development     379   582   961
Sales                      189   257   446
All                        588   882  1470

In [18]:
arr = np.array([ct.iloc[0][0:].values,
                  ct.iloc[1][0:].values])

In [19]:
stats.chi2_contingency(arr)

Chi2ContingencyResult(statistic=1.4709421142834458, pvalue=0.47927963769548776, dof=2, expected_freq=array([[ 24.54785156,  38.45214844,  63.        ],
       [374.45214844, 586.54785156, 961.        ]]))

In [21]:
chi2, p, dof, ex = stats.chi2_contingency(arr)

In [22]:
chi2

1.4709421142834458

In [23]:
p

0.47927963769548776

The pvalue is 0.47927963769548776. Since the pvalue is greater than 0.05, we will NOT reject the Null Hypothesis. 

The Department and Gender columns are not associated with each other as they are independent of each other.